src/eddington/fit_data.py

```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
             return True
         except ValueError:
             return False
+
+    def __make_sheet(self, sheet=None):
+        workbook = openpyxl.Workbook()
+        worksheet = workbook.active
+
+        if sheet:
+            worksheet.title = sheet
+
+        headers = list(self.data.keys())
+        columns = list(self.data.values())
+
+        worksheet.append(headers)
+
+        for row in zip(*columns):
+            worksheet.append(row)
+
+        return workbook, worksheet
+
+    def save_excel(self, output_directory, name=None, sheet=None):
+        """
+        Save :class:`FitData` to xlsx file.
+
+        :param output_directory: str or Path.
+         Path to the directory for the new excel file to be saved.
+        :param name: Optional. The name of the file, without the .xlsx suffix
+        :param sheet: Optional. Name of the sheet that the data will be saved to.
+        """
+        workbook = self.__make_sheet(sheet)[0]
+
+        if name:
+            path = Path(output_directory / f"{name}.xlsx")
+        else:
+            path = Path(output_directory / "FitData.xlsx")
+
+        workbook.save(path)
+
+    def save_csv(self, output_directory, name=None):
+        """
+        Save :class:`FitData` to csv file.
+
+        :param output_directory: str or Path.
+         Path to the directory for the new excel file to be saved.
+        :param name: Optional. The name of the file, without the .csv suffix
+        """
+        worksheet = self.__make_sheet()[1]
```

Comment:
There is no need to make a worksheet here, especially if you drop it emidiatly after.

I would recommend you use [DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) in order to write the csv file.